pkg/api/validation/dynakube/api_url.go

```diff
@@ -77,3 +81,22 @@ func IsThirdGenAPIUrl(_ context.Context, _ *Validator, dk *dynakube.DynaKube) st
 
 	return ""
 }
+
+func IsMutatedApiUrl(ctx context.Context, dv *Validator, dk *dynakube.DynaKube) string {
+	oldDk := &dynakube.DynaKube{}
+	if err := dv.apiReader.Get(ctx, client.ObjectKey{Name: dk.Name, Namespace: dk.Namespace}, oldDk); err != nil {
+		if k8serrors.IsNotFound(err) {
+			return ""
+		}
+
+		log.Info("error occurred while getting dynakube", "err", err.Error())
+
+		return ""
+	}
```

Comment:
instead of doing this

try updating the `ValidateUpdate`, as we already get the "oldObject" there, but we ignored it so far.

The "old" way of running the validators is just kinda outdated, as it never checked for Update specific validations
- the current single list of [validatorErrorFuncs](https://github.com/Dynatrace/dynatrace-operator/blob/9c64c66cd898f972407a55f30a42a4211aaaae7f/pkg/api/validation/dynakube/validation.go#L25) is not going to cut it.

we should have logic like:
```go
func (v *Validator) ValidateUpdate(ctx context.Context, oldObj, newObj runtime.Object) (warnings admission.Warnings, err error) {
	newDk, err := getDynakube(newObj)
	if err != nil {
		return
	}

	oldDk, err := getDynakube(oldObj)
	if err != nil {
		return
	}

	errMessages := v.runValidators(ctx, validatorErrorFuncs, newDk)
	warnings = v.runValidators(ctx, validatorWarningFuncs, newDk)

	errMessages = append(errMessages, v.runUpdateValidators(ctx, updateValidatorFuncs, oldDk, newDk)) // THIS

	if len(errMessages) != 0 {
		err = errors.New(validation.SumErrors(errMessages, "DynaKube"))
	}

	return
}
```

Reply:
I agree, but should this change be part of this, or should we have a refactoring that unifies this behaviour?

My thinking: for features I want the minimal change that makes this feature work within the current patterns.
If we want to change patterns (and I think we should) let's do that in a different PR.

wdyt?